don't fail decoding data from RPM

evgeni opened the pull request:
some versions of the RPM bindings will return strings, some bytes, some
unicode… And only some of these know decode().
So let's just catch that AttributeError and assume there is nothing to
decode if it's raised.

Fixes running certs tools on EL 8.5+ which got a "fixed" RPM.
See https://bugzilla.redhat.com/1840142 for some background.

evgeni committed:
don't fail decoding data from RPM

some versions of the RPM bindings will return strings, some bytes, some
unicode… And only some of these know decode().
So let's just catch that AttributeError and assume there is nothing to
decode if it's raised.

Fixes running certs tools on EL 8.5+ which got a "fixed" RPM.
See https://bugzilla.redhat.com/1840142 for some background.

ekohl commented on katello_certs_tools/katello_ssl_tool.py:
```diff
@@ -611,8 +611,12 @@ def genCaRpm(d, verbosity=0):
-        ver = str(hdr['version'].decode('utf-8'))
-        rel = str(hdr['release'].decode('utf-8'))
+        try:
+            ver = str(hdr['version'].decode('utf-8'))
```
Just thinking out loud: doesn't `.decode()` already return a `str`?

evgeni replied:
maybe, but also trying to keep the least amount of change in ;)

ekohl commented on katello_certs_tools/rhn_rpm.py:
```diff
@@ -75,8 +75,8 @@ def __nonzero__(self):
         if self.hdr[rpm.RPMTAG_FILEDIGESTALGO] \
-           and self.hdr[rpm.RPMTAG_FILEDIGESTALGO].decode('utf-8') in PGPHASHALGO:
-            checksum_type = PGPHASHALGO[self.hdr[rpm.RPMTAG_FILEDIGESTALGO]]
+           and int(self.hdr[rpm.RPMTAG_FILEDIGESTALGO]) in PGPHASHALGO:
```
This int conversion looks like a bugfix. I don't see how this could ever have worked. Looks like a regression introduced in 595c1f85626f5dc50c4f2d7d9e38b4778cc3120f. Prior to that it wasn't decoded at all so was it actually already an int?

evgeni replied:
That's how I read it too, yeah.

evgeni replied:
But if it would be an int, it should have thrown an `AttributeError: 'int' object has no attribute 'decode'`, which I've never seen.

(doesn't increase my liking of this code)

evgeni replied:
this code is unused (and I've removed it now)

explains why it wasn't working but nobody complained ;)

ekohl replied:
Maybe I never looked at it because I looked at #24 to remove the file altogether. It failed but now I finally took the time to make it pass.

evgeni committed:
use latest centos containers from quay

evgeni commented:
@ehelms you asked if the CI would catch this, and #38 shows that it would (at least if running with stream)

evgeni committed:
drop checksum_type

nothing seems to use it since the original import of the code in
996da43f313ad4b500c104a8bbacb85324e4fc27

ehelms commented:
> @ehelms you asked if the CI would catch this, and #38 shows that it would (at least if running with stream)

Awesome

ekohl commented:
Note that #24 showed the tests also failed on current EL8 images.
